technologies/FlexBE/pyrobosim_flexbe_utilities/pyrobosim_flexbe_utilities/world_configuration.py

```diff
@@ -0,0 +1,254 @@
+#!/usr/bin/env python
+
+# Copyright 2024 Christopher Newport University
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""FlexBE utility for loading PyRoboSim world structure."""
+
+import os
+import re
+from collections import deque
+
+from ament_index_python.packages import get_package_share_directory
+
+import numpy as np
+
+import yaml
+
+
+# Helper function: Distance from point to a line segment
+def point_to_line_distance(point, line_start, line_end):
+    """Calculate distance of point to line."""
+    # Convert to NumPy arrays
+    point = np.array(point)
+    line_start = np.array(line_start)
+    line_end = np.array(line_end)
+
+    # Line segment vector
+    line_vec = line_end - line_start
+    # Vector from start of line to the point
+    point_vec = point - line_start
+
+    # Project the point onto the line (normalized)
+    line_len = np.linalg.norm(line_vec)
+    line_unitvec = line_vec / line_len
+    projection = np.dot(point_vec, line_unitvec)
+
+    # Clamp the projection between 0 and line length to handle the endpoints
+    projection = np.clip(projection, 0, line_len)
+
+    # Find the nearest point on the line segment
+    nearest = line_start + projection * line_unitvec
+
+    # Return the Euclidean distance between the point and the nearest point on the line segment
+    return np.linalg.norm(nearest - point)
+
+
+# Helper function:
+def is_point_inside_polygon(point, polygon):
```

Comment:
This works, but if you want, you could consider using the `shapely` package which I use internally in PyRoboSim to do these calculations.